controllers/csiscaleoperator_controller.go

```diff
@@ -891,36 +1108,33 @@ func (r *CSIScaleOperatorReconciler) updateStatus(instance *csiscaleoperator.CSI
 
 			if !r.areAllPodImagesSynced(controllerDeployment, controllerPod) {
 				r.restartControllerPodfromDeployment(logger, controllerDeployment, controllerPod)
+
 			}
 		}
 		phase = csiv1.DriverPhaseCreating
 	}
-	instance.Status.Phase = phase
-	instance.Status.Version = config.DriverVersion
-
-	if !reflect.DeepEqual(originalStatus, instance.Status) {
```

Comment:
Is this if() condition not required anymore ?

Reply:
Not required. This is handled by patch() request in defer function, patch only updates the modified values so no need to check if orginal status and new status are same or not.